File: back/src/main/java/com/LojaVirtual/services/PessoaService.java
package com.LojaVirtual.services;

import java.util.NoSuchElementException;
import java.util.Date;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.LojaVirtual.entities.Pessoa;
import com.LojaVirtual.repositories.PessoaRepository;

@Service
public class PessoaService {

    @Autowired
    private PessoaRepository pessoaRepository;

    public Page<Pessoa> buscarTodos(Pageable pageable) {
        return pessoaRepository.findAll(pageable);
    }

    public Pessoa buscarPorId(Long id) {
        return pessoaRepository.findById(id).get();
    }

    public Pessoa inserir(Pessoa pessoa) {
        pessoa.setDataCriacao(new Date());
        Pessoa pessoaNovo = pessoaRepository.saveAndFlush(pessoa);
        return pessoaNovo;
    }

    public Pessoa alterar(Pessoa pessoa) {
        pessoa.setDataAtualizacao(new Date());
        return pessoaRepository.saveAndFlush(pessoa);
    }

    public void excluir(Long id) {
        Pessoa pessoa = pessoaRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("Pessoa não encontrada."));
        pessoaRepository.delete(pessoa);
    }
}

File: frontendloja/src/services/PessoaService.js
import axios from 'axios';

export class PessoaService{
	URL = "http://localhost:8080/api/pessoa";

	inserir(pessoa){
		return axios.post(this.URL, pessoa);
	}

	alterar(pessoa){
		return axios.put(this.URL, pessoa);
	}

	excluir(id){
		return axios.delete(this.URL+"/"+id);
	}

	listar(page,size){
		return axios.get(this.URL + "?page=" + page + "&size=" + size);
	}
}